src/taco-dependency-installer/test/javaJdkInstaller.ts

```diff
@@ -0,0 +1,344 @@
+                return new nodeFakes.ChildProcess();
+            };
+
+            var expectedTelemetry: TacoUtility.ICommandTelemetryProperties[] = [
+                {
+                    "initialStep.time": { isPii: false, value: "2000" },
+                    step: { isPii: false, value: "initialStep" }
+                },
+                {
+                    "error.description": { isPii: false, value: "ErrorOnChildProcess on attachDmg" },
+                    "install.time": { isPii: false, value: "2000" },
+                    lastStepExecuted: { isPii: false, value: "install" },
+                    step: { isPii: false, value: "install" },
+                    time: { isPii: false, value: "3000" }
+                }
+            ];
+
+            return telemetryGeneratedShouldBe(expectedTelemetry, /Couldn\'t attach dmg file/, done);
+        });
+
+        function fakeExecToTestPackageInstallerFailure(command: string, optionsOrCallback: nodeFakes.ExecSecondArgument,
+            callback: nodeFakes.Callback = null): nodeFakes.IChildProcess {
+            var realCallback = <nodeFakes.Callback> (callback || optionsOrCallback);
+            if (callTime === 0) {
+                // This is for the attach call
+                realCallback(null, /*stdout*/ new Buffer("/Volumes/Macintosh HD"), /*stderr*/ new Buffer(""));
+            } else if (callTime === 1) {
+                // This is for the install package call
+                var couldntRunPackageInstallerError = new Error("Package installer exited with an error");
+                switch (childProcessExecBehavior) {
+                    case ChildProcessExecBehavior.PackageInstallerCommandFailsScenario:
+                        (<any> couldntRunPackageInstallerError).code = 54;
+                        break;
+                    case ChildProcessExecBehavior.PackageInstallerCommandCantBeRunScenario:
+                        break;
+                    case ChildProcessExecBehavior.DetachingDmgFileFailsScenario:
+                        couldntRunPackageInstallerError = null;
+                        break;
+                }
+                realCallback(couldntRunPackageInstallerError, /*stdout*/ new Buffer(""), /*stderr*/ new Buffer(""));
+            } else {
+                // This is for the detach call
+                var error = childProcessExecBehavior === ChildProcessExecBehavior.DetachingDmgFileFailsScenario ?
+                    new Error("Detach dmg command failed to execute") : null;
+                realCallback(error, /*stdout*/ new Buffer(""), /*stderr*/ new Buffer(""));
+            }
+
+            callTime++;
+            return new nodeFakes.ChildProcess();
+        }
+
+        it("generates telemetry error when the package installer gives an error", (done: MochaDone) => {
+            // child_process.exec call will only fail for the install packages call
+            childProcessModule.exec = fakeExecToTestPackageInstallerFailure;
```

Comment:
Nevermind this, I hadn't seen that there is a 4th test near the top of the suite that doesn't use this exec behavior. I instead suggested offline that the `fakeExecToTestPackageInstallerFailure` handler be moved to the top of this suite so that it's above all 4 tests.